spec/amp-framework-hosting.md

```diff
@@ -0,0 +1,234 @@
+
+## Acquire the AMP framework
+
+The AMP framework can be built from source or downloaded pre-built. Building the framework yourself gives you the flexibility of modifying the framework for your needs, or creating your own versioning system. Downloading a pre-built framework means you'll be using an unmodified release, vetted by the AMP Project for distribution.
+
+### Option 1: Build the framework yourself
+
+Refer to the [Developing in AMP](https://github.com/ampproject/amphtml/blob/master/contributing/DEVELOPING.md) guide to familiarize yourself with building and testing the AMP framework. Once you are comfortable with the build system, a few small changes will customize the framework to run from your host.
+
+#### Update URLs config
+
+Modify [`src/config.js`](https://github.com/ampproject/amphtml/blob/master/src/config.js):
+
+- Update the default host in `urls.cdn`. For example, replace  
+  `cdn: env['cdnUrl'] || getMetaUrl('runtime-host') || 'https://cdn.ampproject.org'`
+  with  
+  `cdn: env['cdnUrl'] || getMetaUrl('runtime-host') || 'https://example.com/amp-runtime'`
+- (Optional) Specify an amp-geo fallback API URL in `urls.geoApi`. This API is described in section [amp-geo hotpatching](#amp-geo-hotpatching). For example, replace  
+  `geoApi: env['geoApiUrl'] || getMetaUrl('amp-geo-api')`
+  with  
+  `geoApi: env['geoApiUrl'] || getMetaUrl('amp-geo-api') || 'https://example.com/geo'`
+
+It is not necessary to modify `cdnProxyRegex`, which helps AMP pages identify when they are served from an [AMP Cache](https://amp.dev/documentation/guides-and-tutorials/learn/amp-caches-and-cors/how_amp_pages_are_cached/).
+
+#### Build the framework
+
+Build an AMP release with
+
+```
+gulp dist --config=<config> --version=<version>
+```
+
+where
+
+- `<config>` is one of `canary` or `prod`, indicating whether the build is meant to be served as a canary release or a production release, respectively. When the `--config` flag is omitted, the build system defaults to `prod`.
+- `<version>` is the version you are assigning to the build. When the `--version` flag is omitted, the build system defaults to the commit time of the last commit in the active branch: `TZ=UTC git log -1 --pretty="%cd" --date=format-local:%y%m%d%H%M%S`. Note that the version specified here is not the "runtime version" (RTV), which contains both the config and the version. The runtime version is discussed in more detail in section [Serve the framework](#serve-the-framework).
+
+The built framework can be found in directory `dist`. The version assigned to the build is in `dist/version.txt` and a listing of all files included in build is in `dist/files.txt`. The framework is ready to be moved to and served from your host.
+
+### Option 2: Copy the framework from cdn.ampproject.org
```

Comment:
Sidenote, we should provide a tool for this. See https://github.com/ampproject/amp-toolbox/issues/642

Reply:
A tool would be nice. It would also make sense for the AMP Project to include a bundled `zip` in the Assets of each [release](https://github.com/ampproject/amphtml/releases/). For example, when you look at the [releases of Microsoft Terminal](https://github.com/microsoft/terminal/releases), the source is accompanied by a compiled binary.

Reply:
#27726 mentioned in guide for those who might want to watch the issue.